rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/CharacteristicLongWriteOperation.java

```diff
@@ -209,4 +214,56 @@ public Boolean call(Object emission) {
             }
         };
     }
+
+    private static Func1<Observable<? extends Throwable>, Observable<?>> errorIsRetryableAndAccordingTo(
+            final WriteOperationRetryStrategy writeOperationRetryStrategy,
+            final ByteBuffer byteBuffer,
+            final int batchSize) {
+        return new Func1<Observable<? extends Throwable>, Observable<?>>() {
+            @Override
+            public Observable<?> call(Observable<? extends Throwable> emittedOnWriteFailure) {
+                return emittedOnWriteFailure
+                        .flatMap(toLongWriteFailureOrError())
+                        .compose(writeOperationRetryStrategy)
+                        .doOnNext(repositionByteBufferForRetry());
```

Comment:
This line should go to above the `.compose()` as the user could potentially emit a different `LongWriteFailure` than they were passed which could flaw the reposition of the buffer.